### ISSUE TITLE
`auto` tactic can crash

### ISSUE THREAD

**Issue by peterohanley** [jonsterling/JonPRL#238]
See here (modification of the nats.jonprl example)
https://gist.github.com/peterohanley/98f1cceeab44d774e5b4

It crashes while trying to solve a member(____,U{i}) result with message "subscript out of bounds".

**Comment by jonsterling** [comment]
This is very helpful, thank you! I will try and track this down tonight.

**Comment by jonsterling** [comment]
OK, so I just played around with it, and here's what happened.

The auto tactic discharged more goals than you expected it to, and so when it got to `focus 1 #{...}`, there was no `1st` goal to focus (the indexes start at `0`); that's where the error occurred. We need to invest time into providing really good error messages, since the error message that JonPRL emitted made it look like a bug, and was totally uninformative. I have opened #71 to track that...

Thank you for the report!
